eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/io/JsonNodeToClusteredVariantProcessor.java

```
@@ -0,0 +1,121 @@
+                continue;
+            }
+            assemblyAccession = assemblyInfo.get(0).path("assembly_accession").asText();
+            String alternateAllele, referenceAllele;
+            for (JsonNode alleleAndHgvs : alleleInfo.path("alleles")) {
+                JsonNode spdi = alleleAndHgvs.path("allele").path("spdi");
+                alternateAllele = spdi.path("inserted_sequence").asText();
+                referenceAllele = spdi.path("deleted_sequence").asText();
+                if(alternateAllele.equals(referenceAllele)) {
+                    contig = spdi.path("seq_id").asText();
+                    start = spdi.path("position").asLong(); //dbSNP stores in 0 base, EVA in 1 base
+                    return new ClusteredVariant(assemblyAccession, taxonomyAccession, contig, start,
+                            type, Boolean.FALSE, createdDate);
+                }
+            }
+        }
+        // no PLTP found
+        logger.error("No PLTP found for accession {}", jsonRootNode.path("refsnp_id").asLong());
+        return new ClusteredVariant(assemblyAccession, taxonomyAccession, contig, start,
+                type, Boolean.FALSE, createdDate);
+    }
+
+    /**
+     * @see <a href=https://api.ncbi.nlm.nih.gov/variation/v0/#/>dbSNP JSON 2.0 schema spec</a>
+     * @param variantType DbSNP2.0 JSON variant type
+     * @return EVA variant type
+     */
+    private VariantType prepareVariantType(String variantType) {
+        switch(variantType.toUpperCase()) {
+            case "SNV":
+                return VariantType.SNV;
+            case "MNV":
+                return VariantType.MNV;
+            case "INS":
+                return VariantType.INS;
+            case "DEL":
+                return VariantType.DEL;
+            case "DELINS":
+                return VariantType.INDEL;
+            case "IDENTITY":
+                return VariantType.NO_SEQUENCE_ALTERATION;
+        }
+        return null;
```

We should throw an error here (see [example](https://github.com/EBIvariation/eva-accession/blob/3d675908d14b3d5985a31045a5d8c492ed91089a/eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubSnpNoHgvsToClusteredVariantProcessor.java#L98)) if the variant type does not fall under any of the types that NCBI itself defines.